tests/integration/allocator_accessibility.cpp

```diff
@@ -82,7 +82,10 @@ struct allocate_and_use<cuda_platform> {
   {
     size_t* data{do_allocate(alloc, size * sizeof(size_t))};
     tester<<<1, 16>>>(data, size);
-    cudaDeviceSynchronize();
+    cudaError_t err = cudaDeviceSynchronize();
+    if (err != cudaSuccess) {
+      UMPIRE_ERROR(umpire::runtime_error, fmt::format("Error when trying to sync device: {}", cudaGetErrorString(err)));
```

Comment:
For consistency, "Error when trying to sync CUDA device: {}". Though if you wanted to leave off CUDA or HIP in all these error messages, I would be fine with that.